scootapi/server/runJob.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"fmt"
+	"github.com/nu7hatch/gouuid"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/scheduler"
+	"github.com/scootdev/scoot/scootapi/gen-go/scoot"
+)
+
+// Implementation of the RunJob API
+func runJob(scheduler scheduler.Scheduler, def *scoot.JobDefinition) (*scoot.JobId, error) {
+
+	jobDef, err := thriftJobToScoot(def)
+	// TODO: change to return scoot.NewInvalidRequest()
+	if err != nil {
+		return nil, err
+	}
+
+	err = validateJob(jobDef)
+	if err != nil {
+		return nil, err
+	}
+
+	job := sched.Job{
+		Id:  generateJobId(),
+		Def: jobDef,
+	}
+
+	err = scheduler.ScheduleJob(job)
```

Comment:
How do you think about generating the job ID in the scheduler vs. the server package?

I think it's weird for the scheduler to have to think about "what if a client gives me duplicate job ids", so I'd consider moving it into there. (Perhaps even into the saga log? Like, StartSaga could create the new ID, which might be totally natural within certain implementations of Saga)